### PR TITLE
Rename UserNotifier to UserMailer

### LINKED ISSUE #3201: Rename UserNotifier to UserMailer
As the title says, rename `UserNotifier` to `UserMailer`, and all of the various related files and method calls.

### PR DESCRIPTION
**Pull request by hahattan**
In your PR did you:

  - [x] Include a description of the changes?
  - [x] Mention the issue the PR addresses?
  - [ ] Include screenshots of any changes to the UI?
  - [x] Isolate any changes to gems (meaning that any new, updated, or removed gems and resulting code changes should be in their own PR)?
  - [x] Add and/or update specs for your code?

fix #3201

### CODE REVIEW

File: app/mailers/user_mailer.rb [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
   def send_followers_email(users, this_case)
     @this_case = this_case
     users.each do |user|
-      Rails.logger.info("UserNotifier#send_followers_email: Sending notification to #{user.email} about case #{this_case.title}")
+      Rails.logger.info("UserMailer#send_followers_email: Sending notification to #{user.email} about case #{this_case.title}")
       mail(to: user.email, subject: "The #{@this_case.title} case has been updated on EBWiki.")
     end
   end
 
   def send_deletion_email(users, this_case)
```
Comment (rlgreen91) [inline app/mailers/user_mailer.rb]: Same as above please.

File: doc/app/ArticlesController.html [hunk 2]
```diff
@@ -188,7 +188,7 @@ <h3>Public Instance Methods</h3>
 <span class="ruby-keyword">def</span> <span class="ruby-identifier">destroy</span>
   <span class="ruby-ivar">@article</span>.<span class="ruby-identifier">destroy</span>
   <span class="ruby-identifier">flash</span>[<span class="ruby-value">:success</span>] = <span class="ruby-node">&quot;Article was removed! #{make_undo_link}&quot;</span>
-  <span class="ruby-constant">UserNotifier</span>.<span class="ruby-identifier">notify_of_removal</span>(<span class="ruby-ivar">@article</span>.<span class="ruby-identifier">followers</span>,<span class="ruby-ivar">@article</span>).<span class="ruby-identifier">deliver_now</span>
+  <span class="ruby-constant">UserMailer</span>.<span class="ruby-identifier">notify_of_removal</span>(<span class="ruby-ivar">@article</span>.<span class="ruby-identifier">followers</span>,<span class="ruby-ivar">@article</span>).<span class="ruby-identifier">deliver_now</span>
```
Comment (rlgreen91) [inline doc/app/ArticlesController.html]: You didn't do these changes manually, did you?  Anything in the `doc` directory can be ignored, honestly.

File: spec/mailers/previews/user_mailer_preview.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
-# frozen_string_literal: true
-
-# Preview all emails at http://localhost:3000/rails/mailers/user_notifier
-class UserNotifierPreview < ActionMailer::Preview
-  def send_followers_email
-    UserNotifier.send_followers_email([User.first], Case.last)
-  end
-end
+# frozen_string_literal: true
```
Comment (rlgreen91) [inline spec/mailers/previews/user_mailer_preview.rb]: Please add previews for the other actions as well.

File: doc/app/ArticlesController.html (hunk 2 above)
Comment (hahattan) [reply doc/app/ArticlesController.html]: I actually do these manually .. so I need to undo all the changes in the `doc` directory ?

Comment (rlgreen91) [reply doc/app/ArticlesController.html]: Oh no, you can leave them there.  I was just saying that you didn't have to do them.

**Comment by rlgreen91** [comment]
Hi @hahattan, are the tests passing for you locally?  I see where you changed the method definitions to use keyword arguments (thanks for that) but it doesn't seem like you updated the methods to pass keyword arguments when you call them in a controller.  I don't think the tests would pass if you'd run them.

Once you do that, I'll go ahead and approve and merge.

**Comment by hahattan** [comment]
> Hi @hahattan, are the tests passing for you locally? I see where you changed the method definitions to use keyword arguments (thanks for that) but it doesn't seem like you updated the methods to pass keyword arguments when you call them in a controller. I don't think the tests would pass if you'd run them.
> 
> Once you do that, I'll go ahead and approve and merge.

Hi, how do I fix ArgumentError when run tests locally ?
`Missing required arguments: aws_access_key_id, aws_secret_access_key`

**Comment by rlgreen91** [comment]
@hahattan Don't worry about running the tests locally then, I'll just handle that myself.  If you update the method calls in the controller to use keyword arguments, then I can verify that everything works and merge the PR.

**Comment by hahattan** [comment]
Hi, I've tried to update the controller and sync with the master branch, please check if tests run well

**Comment by rlgreen91** [comment]
@hahattan I was able to resolve the merge conflicts and test everything, it looks great.  Can you give me push access to your fork?  That way I can update your branch with the resolved conflicts and merge from there.

**Comment by hahattan** [comment]
@rlgreen91 hi I've added you to the collaborators

**Comment by rlgreen91** [comment]
> @rlgreen91 hi I've added you to the collaborators

Thanks!  For whatever weird reason, it wouldn't let me push to your branch, just master.  Can you create a new PR from your master branch?  Then I can approve and merge it.

**Comment by hahattan** [comment]
I can't because my master branch is identical to upstream(yours)

**Comment by rlgreen91** [comment]
Ok, I'm not sure how it missed the ref, but I was able to fix it, so I'll merge shortly.  Thanks again for contributing!

**Pull request merged**